pkg/mpc-examples/contracts/00-null-controller/NullControllerFactory.sol

```diff
@@ -0,0 +1,124 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/IManagedPool.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Create2.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "../interfaces/IManagedPoolFactory.sol";
+import "./NullController.sol";
+
+/**
+ * @title NullControllerFactory
+ * @notice Factory for a Managed Pool and NullController.
+ * @dev Determines controller deployment address, deploys pool (w/ controller address as argument), then controller.
+ */
+contract NullControllerFactory is Ownable {
+    mapping(address => bool) public isControllerFromFactory;
+
+    address public immutable managedPoolFactory;
+    IVault public immutable balancerVault;
+    bool public isDisabled;
+
+    uint256 private _nextControllerSalt;
+    address private _lastCreatedPool;
+
+    // This struct is a subset of IManagedPoolFactory.NewPoolParams which omits arguments
+    // that this factory will override and are therefore unnecessary to provide. It will
+    // ultimately be used to populate IManagedPoolFactory.NewPoolParams.
+    struct MinimalPoolParams {
+        string name;
+        string symbol;
+        IERC20[] tokens;
+        uint256[] normalizedWeights;
+        uint256 swapFeePercentage;
+        bool swapEnabledOnStart;
+        uint256 managementAumFeePercentage;
+        uint256 aumFeeId;
+    }
+
+    event ControllerCreated(address indexed controller, bytes32 poolId);
+    event Disabled();
+
+    constructor(IVault vault, address factory) {
+        balancerVault = vault;
+        managedPoolFactory = factory;
+    }
+
+    /**
+     * @dev Return the address of the most recently created pool.
+     */
+    function getLastCreatedPool() external view returns (address) {
+        return _lastCreatedPool;
+    }
+
+    function create(MinimalPoolParams memory minimalParams) external {
+        require(!isDisabled, "Controller factory disabled");
+        require(!IManagedPoolFactory(managedPoolFactory).isDisabled(), "Pool factory disabled");
+
+        bytes32 controllerSalt = bytes32(_nextControllerSalt);
+        _nextControllerSalt += 1;
+
+        bytes memory controllerCreationCode = abi.encodePacked(
+            type(NullController).creationCode,
+            abi.encode(balancerVault)
+        );
+        address expectedControllerAddress = Create2.computeAddress(controllerSalt, keccak256(controllerCreationCode));
+
+        // build arguments to deploy pool from factory
```

Comment:
Okay I'm gonna start to get really nit-picky with comment formatting since this factory will probably be copied all over the place.
```suggestion
        // Build arguments to deploy pool from factory.
```